### PR TITLE
[GUI] Guard shielded flows pre-v5 enforcement

### PR DESCRIPTION
**Pull request by furszy**
Visual notification, do not try to create shielded addresses or process shielded spends if v5 upgrade hasn't been enforced yet.

### CODE REVIEW

File: src/qt/pivx/send.cpp [hunk 1]
```diff
@@ -394,9 +394,16 @@ void SendWidget::ProcessSend(const QList<SendCoinsRecipient>& recipients, bool h
 {
     // First check SPORK_20 (before unlock)
     bool isShieldedTx = hasShieldedOutput || !isTransparent;
-    if (isShieldedTx && walletModel->isSaplingInMaintenance()) {
-        inform(tr("Sapling Protocol temporarily in maintenance. Shielded transactions disabled (SPORK 20)"));
-        return;
+    if (isShieldedTx) {
+        if (!walletModel->isSaplingEnforced()) {
+            inform(tr("Cannot perform shielded operations, v5 upgrade hasn't being enforced yet!"));
```
Comment (Fuzzbawls) [inline src/qt/pivx/send.cpp:399]: change to either `hasn't been enforced` or `isn't being enforced`

File: src/qt/pivx/send.cpp [hunk 2]
```diff
@@ -667,6 +674,11 @@ void SendWidget::onCoinControlClicked()
 
 void SendWidget::onShieldCoinsClicked()
 {
+    if (!walletModel->isSaplingEnforced()) {
+        inform(tr("Cannot perform shielded operations, v5 upgrade hasn't being enforced yet!"));
```
Comment (Fuzzbawls) [inline src/qt/pivx/send.cpp:678]: change to either `hasn't been enforced` or `isn't being enforced`

File: src/qt/pivx/send.cpp [hunk 3]
```diff
@@ -751,6 +763,13 @@ void SendWidget::onCheckBoxChanged()
 void SendWidget::onPIVSelected(bool _isTransparent)
 {
     isTransparent = _isTransparent;
+
+    if (!isTransparent && !walletModel->isSaplingEnforced()) {
+        ui->pushLeft->setChecked(true);
+        inform(tr("Cannot perform shielded operations, v5 upgrade hasn't being enforced yet!"));
```
Comment (Fuzzbawls) [inline src/qt/pivx/send.cpp:769]: change to either `hasn't been enforced` or `isn't being enforced`

File: src/qt/pivx/receivewidget.cpp [hunk 4]
```diff
@@ -338,7 +338,14 @@ void ReceiveWidget::sortAddresses()
 
 void ReceiveWidget::onTransparentSelected(bool transparentSelected)
 {
-    this->shieldedMode = !transparentSelected;
+    shieldedMode = !transparentSelected;
+
+    if (shieldedMode && !walletModel->isSaplingEnforced()) {
+        ui->pushLeft->setChecked(true);
+        inform(tr("Cannot perform shielded operations, v5 upgrade hasn't being enforced yet!"));
```
Comment (Fuzzbawls) [inline src/qt/pivx/receivewidget.cpp:345]: change to either `hasn't been enforced` or `isn't being enforced`

**Review by Fuzzbawls — CHANGES_REQUESTED** [review]
grammar changes ;)

File: src/qt/pivx/send.cpp (hunk 1 above)
Comment (furszy) [reply src/qt/pivx/send.cpp:399]: typo fixed

File: src/qt/pivx/send.cpp (hunk 2 above)
Comment (furszy) [reply src/qt/pivx/send.cpp]: typo fixed

File: src/qt/pivx/send.cpp (hunk 3 above)
Comment (furszy) [reply src/qt/pivx/send.cpp]: typo fixed

File: src/qt/pivx/receivewidget.cpp (hunk 4 above)
Comment (furszy) [reply src/qt/pivx/receivewidget.cpp]: typo fixed

**Review by random-zebra — APPROVED** [review]
utACK 1c43a1eac8634c3c43b4a79ab36a9f52f8643d98

**Review by Fuzzbawls — APPROVED** [review]
ACK 1c43a1eac8634c3c43b4a79ab36a9f52f8643d98

**Pull request merged**